packages/formatters/src/formatters.ts

```diff
@@ -373,6 +373,87 @@ export function messageLink<C extends Snowflake, M extends Snowflake, G extends
 	return `${guildId === undefined ? channelLink(channelId) : channelLink(channelId, guildId)}/${messageId}`;
 }
 
+export enum HeadingLevel {
+	One = 1,
+	Two,
+	Three,
+}
+
+/**
+ * Formats the content into a heading level.
+ *
+ * @typeParam C - This is inferred by the supplied content
+ * @param content - The content to wrap
+ * @param level - The heading level
+ */
+export function heading<C extends string>(content: C, level?: HeadingLevel.One): `# ${C}`;
+
+/**
+ * Formats the content into a heading level.
+ *
+ * @typeParam C - This is inferred by the supplied content
+ * @param content - The content to wrap
+ * @param level - The heading level
+ */
+export function heading<C extends string>(content: C, level: HeadingLevel.Two): `## ${C}`;
+
+/**
+ * Formats the content into a heading level.
+ *
+ * @typeParam C - This is inferred by the supplied content
+ * @param content - The content to wrap
+ * @param level - The heading level
+ */
+export function heading<C extends string>(content: C, level: HeadingLevel.Three): `### ${C}`;
+
+export function heading(content: string, level?: HeadingLevel) {
+	switch (level) {
+		case HeadingLevel.Three:
+			return `### ${content}`;
+		case HeadingLevel.Two:
+			return `## ${content}`;
+		default:
+			return `# ${content}`;
+	}
+}
+
+export type RecursiveArray<T> = readonly (RecursiveArray<T> | T)[];
```

Comment:
Basic documentation should be added to this, even if it's as simple as "A type that recursively traverses into arrays."